src/rules/tsUtils.ts

```diff
@@ -28,6 +43,26 @@ export const isFunction = (node: TSESTree.Node): node is FunctionExpression =>
   node.type === AST_NODE_TYPES.FunctionExpression ||
   node.type === AST_NODE_TYPES.ArrowFunctionExpression;
 
+export const isHook = (node: TSESTree.CallExpression): boolean => {
```

Comment:
I wonder if we can make this (and the others) into a type guard. Sort of like what I've done in https://github.com/jest-community/eslint-plugin-jest/blob/6958f04ce09bee8988b84453b3e84e9358d2482c/src/rules/lowercase-name.ts#L7-L51

It narrows down the types as we go

(might wanna move the ones I linked to in here as well?)

Reply:
I'd advise against that - it's actually something I've recently been talking about with a few different people.

The problem is that type guards imply if something is *or is not* a given type, which will cause massive problems if you start putting business logic in your guards.

For example:

```
const isPositiveNumber = (n: unknown): n is number => typeof n === 'number' && n > 0;

const fn = (p: number | string): string => {
  if(isPositiveNumber(p)) {
    return p.toString();
  } //        ^ type: number

  return p.toUpperCase();
  //        ^ type: string | !number
}
```

Another way (and maybe more accurate) to say is it that type guards don't just narrow, they also *negate*.

Reply:
However in looking over your code, I think you might already know that, and has accounted for it :)

If you're happy with it, I'm happy with it :joy:

Reply:
Yeah, they should definitely be logic free, just checking the _type_ of things, not if they pass some constraint. But e.g. verifying the name of a function and the type of its arguments (or at least, say, the first n arguments) is really useful as we then don't have to use type casts to please the compiler. As long as we're careful to just do narrowing that types can verify, I think we're fine

Reply:
Yeah totally - your code I think is perfect; it wouldn't work w/ the `number` example (that needs [this](https://github.com/Microsoft/TypeScript/issues/15480)), but it should be totally spot on for this.

The only thing I think is that the enums will need to be strings:

```
enum DescribeAlias {
  'describe' = 'describe',
  'fdescribe' = 'fdescribe',
  'xdescribe' = 'xdescribe',
}
```

Otherwise, we can't use them as both type values and as enums, since `keyof DescribeAlias` currently returns `0,1,2,etc` :(

Reply:
That's fine 🙂 Not sure what the cleanest is there, but easy enough to change if a better alternative comes up at some point

Reply:
I've pushed up an implementation, which I am actually really happy w/.

Despite how it might first appear, I think it shuffles around the current learning curve rather than adding to it, while making it super quick to implement scoping on rules to target specific jest functions :D

Let me know what you think - I've applied it to `lowercase-name` as a showcase, but can cherry-pick that into another PR if you'd like.